tensorboard/components/experimental/plugin_lib/core.ts

```diff
@@ -23,3 +23,11 @@ import {sendMessage} from './plugin-guest';
 export async function getURLPluginData() {
   return sendMessage('experimental.GetURLPluginData');
 }
+
+/**
+ * Listen to when data reloads on TensorBoard. There are two ways in which
+ * data reloads--(1) automatic refresh and (2) user clicks on a reload button.
```

Comment:
I wonder whether there is any change this might trigger on initial load as well.

After /plugins_listing succeeds, CoreEffects dispatches `pluginsListingLoaded`:
(A)
Something creates + registers the plugin's iframe.  In possibly another thread, the iframe starts listening on a MessageChannel.

(B)
Reducer receives `pluginsListingLoaded`, and updates the timestamp.  CoreHostImpl subscribes to changes in the timestamp, and broadcasts a message to any registered iframes.

Are we guaranteed that (A) will always happen after (B)?

Reply:
I must admit, I don't get the concern. Currently contract is that we will call the callback on data reload, not on initial load. I think you are talking about initial load and this callback getting invoked there? 

I expect plugins to read the initial data using a necessary API on its initial load. Kind of like how `setOnRunsChanged` vs. `getRuns` are two separate APIs while `setOnRunsChanged` already provides you with the list of runs.

Reply:
> Currently contract is that we will call the callback on data reload, not on initial load.

That is a reasonable contract, but is that contract upheld in this PR?  Looking at https://github.com/tensorflow/tensorboard/blob/master/tensorboard/webapp/core/effects/core_effects.ts#L81, the initial `coreLoaded` will also trigger pluginsListingLoaded.

How do we know that the initial load does not cause this API to fire?

Reply:
There is a difference between:
1. TensorBoard data loaded for the first time vs.
2. Plugin freshly stamped. 

The contract is that we will trigger the callback when data ever reloads which includes (1).